tiatoolbox/tools/stainaugment.py

```diff
@@ -0,0 +1,216 @@
+            (`alpha` belonging to [1-sigma1, 1+sigma1] range). Default is 0.5.
+        sigma2 (float): Controls the extent of the stain concentrations shift parameter
+            (`beta` belonging to [-sigma2, sigma2] range). Default is 0.25.
+        augment_background (bool): Specifies whether to apply stain augmentation on the
+            background or not. Default is False, which indicates that only tissue region
+            will be stain augmented.
+        always_apply (False): For use with 'albumentations' pipeline. Please refer to
+            albumentations documentations for more information.
+        p (0.5): For use with 'albumentations' pipeline which specifies the probability
+            of using the augmentation in a 'albumentations' pipeline. . Please refer to
+            albumentations documentations for more information.
+
+    Attributes:
+        stain_normaliser: Fitted stain normalization class.
+        stain_matrix (:class:`numpy.ndarray`): extracted stain matrix from the image
+        source_concentrations (:class:`numpy.ndarray`): extracted stain
+            concentrations from the input image.
+        n_stains (int): number of stain channels in the stain concentrations.
+            Expected to be 2 for H&E stained images.
+        tissue_mask (:class:`numpy.ndarray`): tissue region mask in the image.
+
+    Examples:
+        >>> '''Using the stain augmentor in the 'albumentations' pipeline'''
+        >>> from tiatoolbox.tools.stainaugment import StainAugmentaiton
+        >>> import albumentations as A
+        >>> # Defining an examplar stain matrix as refrence
+        >>> stain_matrix = np.array([[0.91633014, -0.20408072, -0.34451435],
+        ...                [0.17669817, 0.92528011, 0.33561059]])
+        >>> # Define albumentations pipeline
+        >>> aug_pipline = A.Compose([
+        ...                         A.RandomRotate90(),
+        ...                         A.Flip(),
+        ...                         StainAugmentaiton(stain_matrix=stain_matrix)
+        ...                         ])
+        >>> # apply the albumentations pipeline on an image (RGB numpy unit8 type)
+        >>> img_aug = aug(image=img)['image']
+
+        >>> '''Using the stain augmentor stand alone'''
+        >>> from tiatoolbox.tools.stainaugment import StainAugmentaiton
+        >>> # Defining an examplar stain matrix as refrence
+        >>> stain_matrix = np.array([[0.91633014, -0.20408072, -0.34451435],
+        ...                [0.17669817, 0.92528011, 0.33561059]])
+        >>> # Instantiate the stain augmentor and fit it on an image
+        >>> stain_augmentor = StainAugmentation(stain_matrix=stain_matrix)
+        >>> stain_augmentor.fit(img)
+        >>> # Now using the fitted `stain_augmentor` in a loop to generate
+        >>> # several augmented instances from the same image.
+        >>> for i in range(10):
+        ...     img_aug = stain_augmentor.augment()
+    """
```

Comment:
new line

Reply:
all docstring issues are addressed in 26dcf3f.
Thanks Shan.